functors task (1)/student/src/impl.cpp

```diff
@@ -1,2 +1,16 @@
 #include "impl.h"
 
+static constexpr BeerBrand kBeerBrandsOrdered[] = {
```

Comment:
**never use c array in C++**
use std::array instead!

Reply:
Never say never :)
There are some pros and cons of using std::array vs C-array.
1. std::array must specify size, it is not flexible, but yes, it is much better than do "sizeof() / sizeof()", my bad... until C++17.
2. Both can be passed to algorithms as they have std::begin() and std::end().
3. Both can be passed to other [template] functions.
4. constexpr int array[] = { 1, 2, ..., N }; looks much simpler and doesn't suffer from (1).
In general, I'd say that from C++17 (where we've got std::size() for both of them) there are not much advantages in std::array to declare const[expr] arrays... They are quite similar.

Anyway, this one was removed.

Reply:
You must specify size for any kind of array, because VLA is not C++ way.
But consider advantages of std::array (it is not worse than a c array):
* You can always access size
* Value semantics
* No casts to a pointer
* Range checking